### DISCUSSION TITLE
How to correctly setting path_list ?

### DISCUSSION THREAD

**Discussion by xathrya** [RustPython/RustPython#5171] [General]
I want to use a custom path to host my libs which then will be used from the script.

In rust, I use following code.

```rust
let mut settings = rustpython_vm::Settings::default();
settings.path_list.push("/tmp/mylib".to_owned());

rustpython_vm::Interpreter::with_init(settings, |vm| {
    vm.add_native_modules(rustpython_stdlib::get_module_inits());
})
.enter(|vm| {
    let scope = vm.new_scope_with_builtins();
    let code_obj = vm
        .compile(source, rustpython_vm::compiler::Mode::Exec, "<embedded>".to_owned())
        .map_err(|err| vm.new_syntax_error(&err, Some(source)))?;

    let _ = vm.run_code_obj(code_obj, scope);

    Ok(())
});
```

Inside the /tmp/mylib I have `greet.py` with following content:

```python
def hello():
    print("Hello World!")
```

the script that will run 

```python
import greet

greet.hello()
```

but it gives error which said the module is not found

```
Traceback (most recent call last):
  File "<embedded>", line 1, in <module>
ModuleNotFoundError: No module named 'greet'
```

Did I miss something?

**Comment by youknowone** [comment]
It looks fine to me. What `sys.path` returns before `import greet`?

**Comment by xathrya** [comment]
Hi, sorry for late reply
Here's the updated script.

```python
import sys
print(sys.path)

import greet
greet.hello()
```

It print 

```
['/tmp/mylib']
Traceback (most recent call last):
  File "<embedded>", line 16, in <module>
ModuleNotFoundError: No module named 'greet'
```

**Comment by xathrya** [comment]
bumping this, any idea why it gives error?

**Reply by youknowone** [reply]
Is `greet.py` placed in `/tmp/mylib`, but it doesn't work? I have no idea in that case. That's expected to be working

**Reply by xathrya** [reply]
yes, `greet.py` is inside the `/tmp/mylib`.

I forgot to say that error message is obtained by printing exception

```rust
match vm.run_code_obj(code_obj, scope) {
            Ok(_)   => {},
            Err(e)  => {
                vm.print_exception(e);
            }
        }
```

I  use mac, the rustpython-vm is 0.3.1

**Reply by youknowone** [reply]
I am sorry, I was forgetting about this discussion. Probably missing `importlib` feature on rustpython-vm.

#5741

**Reply by xathrya** [reply]
oh great, it was because of importlib!
